### PR TITLE
[Llama4] Attention temperature tuning for  NoRoPE layers

### PR DESCRIPTION
**Pull request by gagika**
# Description

[Llama4] Attention temperature tuning for  No RoPE layers

b/409848691

# Tests

Local test in Jax (logits tests are pending)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/layers/attentions.py [hunk 1]
```diff
@@ -1317,6 +1322,14 @@ def __call__(
     if self.query_pre_attn_scalar and self.query_pre_attn_scalar != 1.0:
       query = query * self.query_pre_attn_scalar
 
+    if self.temperature_tuning and not self.use_rope:
+      attn_scales = (
+          jnp.log(jnp.floor((inputs_positions + 1.0) / self.temperature_tuning_floor_scale) + 1.0)
```
Comment (jrplatin) [inline MaxText/layers/attentions.py:1327]: Is `inputs_positions` something you want to cast to `float`? I noticed [HF](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama4/modeling_llama4.py#L333) is doing that but not sure if it's needed or not

**Review by jrplatin — APPROVED** [review]
LGTM!

File: MaxText/layers/attentions.py (hunk 1 above)
Comment (gagika) [reply MaxText/layers/attentions.py]: Done, explicitly converting to `inputs_positions.astype(self.dtype)`.

**Review by khatwanimohit — APPROVED** [review]

**Review by richjames0 — APPROVED** [review]
nit: the scope change for rmsnorm is a separate thing and would ideally be a separate PR. maybe you could change the title of the PR to reflect?

File: MaxText/layers/attentions.py [hunk 2]
```diff
@@ -1281,24 +1285,25 @@ def __call__(
...
-
-        key = RMSNorm(
-            dtype=self.config.dtype,
-            weight_dtype=self.config.weight_dtype,
-            name="key_norm",
-            epsilon=self.config.normalization_layer_epsilon,
-            kernel_axes=("norm",),
-        )(key)
+
+    is_llama4_decoder_block = self.config.decoder_block == "llama4"
```
Comment (gobbleturk) [inline MaxText/layers/attentions.py:1289]: To confirm understanding this PR also changes the behavior with fused_qkv - now RMSNorm will be applied whereas previously it wasn't? We probably don't have any correctness tests covering fused_qkv

**Review by gobbleturk — APPROVED** [review]

File: MaxText/layers/attentions.py (hunk 2 above)
Comment (gagika) [reply MaxText/layers/attentions.py:1289]: yes, sorry, I missed this comment, this fixes a bug for Llama 4 / gemma 3 with fused_qkv,
We will need to add some correctness tests with fused_qkv too.
